apps/studio/components/layouts/TableEditorLayout/TableEditorMenu.tsx

```diff
@@ -193,113 +197,113 @@ const TableEditorMenu = () => {
+                      Entity Type
+                    </DropdownMenuRadioItem>
+                  </DropdownMenuRadioGroup>
+                </DropdownMenuContent>
+              </DropdownMenu>
 
-                      <button
-                        className="cursor-pointer text-foreground-lighter transition-colors hover:text-foreground"
-                        onClick={refreshTables}
-                      >
-                        <IconRefreshCw className={isRefetching ? 'animate-spin' : ''} size={14} />
-                      </button>
-                    </div>
-                  </div>
-                </>
-              }
-            />
+              <button
+                className="cursor-pointer text-foreground-lighter transition-colors hover:text-foreground"
+                onClick={refreshTables}
+              >
+                <IconRefreshCw className={isRefetching ? 'animate-spin' : ''} size={14} />
+              </button>
+            </div>
+          </div>
 
-            <div className="flex flex-1">
-              <InfiniteList
-                items={entityTypes}
-                ItemComponent={EntityListItem}
-                itemProps={{
-                  projectRef: project?.ref,
-                  id: Number(id),
-                }}
-                getItemSize={() => 28}
-                hasNextPage={hasNextPage}
-                isLoadingNextPage={isFetchingNextPage}
-                onLoadNextPage={() => fetchNextPage()}
-              />
+          {isLoading && (
+            <div className="flex flex-col px-2 gap-1 pb-4">
+              <ShimmeringLoader className="w-full h-7 rounded-md" delayIndex={0} />
+              <ShimmeringLoader className="w-full h-7 rounded-md" delayIndex={1} />
+              <ShimmeringLoader className="w-full h-7 rounded-md" delayIndex={2} />
             </div>
-          </Menu>
-        )}
+          )}
+
+          {isError && (
+            <AlertError error={(error ?? null) as any} subject="Failed to retrieve tables" />
+          )}
+
+          {isSuccess && (
+            <>
+              {searchText.length === 0 && (entityTypes?.length ?? 0) <= 0 && (
+                <div className="mx-2 my-2 space-y-1 rounded-md border border-muted bg-surface-100 py-3 px-4">
+                  <p className="text-xs">No entities available</p>
+                  <p className="text-xs text-foreground-light">
+                    This schema has no entities available yet
+                  </p>
+                </div>
+              )}
+              {searchText.length > 0 && (entityTypes?.length ?? 0) <= 0 && (
+                <div className="mx-2 my-2 space-y-1 rounded-md border border-muted bg-surface-100 py-3 px-4">
+                  <p className="text-xs">No results found</p>
+                  <p className="text-xs text-foreground-light">
+                    There are no entities that match your search
```

Comment:
let's do this instead

`Your search for "${searchText}" did not return any results`